staging/src/k8s.io/dynamic-resource-allocation/structured/allocator_test.go

```diff
@@ -3464,6 +3562,382 @@ func TestAllocator(t *testing.T) {
 				deviceAllocationResult(req1, driverA, pool1, device1, false),
 			)},
 		},
+		"consumable-capacity-multi-allocatable-device-with-consumable-capacity": {
+			features: Features{
+				ConsumableCapacity: true,
+			},
+			claimsToAllocate: objects(
+				claimWithCapacityReqeust(claim0, req0, classA, 1, capacityRequests(&one)),
```

Comment:
There is a typo in the name of the `claimWithCapacityReqeust` function. It should be `claimWithCapacityRequest`.

But I'm wondering if we could avoid adding such a specific function by leveraging the `wrap...` pattern here. It doesn't seem like we have types for `wrapDeviceRequest` or `wrapDeviceSubRequest`, but maybe using those would simplify this? I see you have already done something like this for the device type in the `ResourceSlice`.